scheduler/reconnecting_picker/reconnecting_picker_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package reconnectingpicker
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/shoenig/test/must"
+)
+
+func TestPickReconnectingAlloc_NewerVersion(t *testing.T) {
+	rp := New(nil)
+	ds := &structs.DisconnectStrategy{
+		Reconcile: "best-score",
+	}
+	original := &structs.Allocation{
+		Job: &structs.Job{
+			Version:     1,
+			CreateIndex: 1,
+		},
+	}
+	replacement := &structs.Allocation{
+		Job: &structs.Job{
+			Version:     2,
+			CreateIndex: 2,
+		},
+	}
```

Comment:
It may be useful to matrix test each field being modified independently. Testing for a change in both can hide bugs in one of the two checks.